### PR TITLE
*: make rust-clippy happy

### PR DESCRIPTION
**Pull request by ngaut**

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

File: tests/test_raft.rs [hunk 1]
```diff
@@ -1415,22 +1415,22 @@ fn test_leader_election_with_check_quorum() {
     // we can not let system choosing the value of randomizedElectionTimeout
     // otherwise it will introduce some uncertainty into this test case
     // we need to ensure randomizedElectionTimeout > electionTimeout here
-    let a_election_timeout = nt.peers.get(&1).unwrap().get_election_timeout();
-    let b_election_timeout = nt.peers.get(&2).unwrap().get_election_timeout();
+    let a_election_timeout = &nt.peers[&1].get_election_timeout();
```
Comment (BusyJay) [inline tests/test_raft.rs]: `&` is redundant.

File: tests/test_raft.rs [hunk 2]
```diff
@@ -1415,22 +1415,22 @@ fn test_leader_election_with_check_quorum() {
...
     nt.peers.get_mut(&1).unwrap().set_randomized_election_timeout(a_election_timeout + 1);
     nt.peers.get_mut(&2).unwrap().set_randomized_election_timeout(b_election_timeout + 2);
 
     // Immediately after creation, votes are cast regardless of the election timeout
 
     nt.send(vec![new_message(1, 1, MessageType::MsgHup, 0)]);
 
-    assert_eq!(nt.peers.get(&1).unwrap().state, StateRole::Leader);
-    assert_eq!(nt.peers.get(&3).unwrap().state, StateRole::Follower);
+    assert_eq!(&nt.peers[&1].state, StateRole::Leader);
```
Comment (BusyJay) [inline tests/test_raft.rs]: `&` is redundant.

**Comment by ngaut** [comment]
PTAL

File: tests/test_raft.rs [hunk 3]
```diff
@@ -2507,7 +2504,7 @@ fn test_leader_transfer_second_transfer_to_another_node() {
     // Transfer leadership to another node.
     nt.send(vec![new_message(2, 1, MessageType::MsgTransferLeader, 0)]);
 
-    check_leader_transfer_state(nt.peers.get(&1).unwrap(), StateRole::Follower, 2);
+    check_leader_transfer_state(&nt.peers[&1], StateRole::Follower, 2);
```
Comment (BusyJay) [inline tests/test_raft.rs]: `&` is redundant.

Comment (BusyJay) [inline tests/test_raft.rs]: Ignore it, my mistake.

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**